tests/unit/s2n_tls13_pq_handshake_test.c

```diff
@@ -474,9 +579,15 @@ int main()
         bool len_prefix_expected = vector->len_prefix_expected;
 
         if (!s2n_pq_is_enabled()) {
-            /* If PQ is disabled, we always expected to negotiate ECC. */
+            /* If PQ is disabled, for older policies we expect to negotiate x25519 if available.
+             * For newer policies, we always expect to negotiate NIST curves on ECC-only fallback.
+             */
             kem_group = NULL;
-            curve = expected_curve;
+            if (server_policy == &security_policy_pq_tls_1_3_2023_06_01) {
+                curve = &s2n_ecc_curve_secp256r1;
```

Comment:
I'm not sure I understand this branching. Why can't you just change the expected curve?

Reply:
I tried to explain the reason for this in the above comment. The reason for this is that older PQ security policies will end up negotiating an x25519 ECC fallback, but the new security policy doesn't include non-NIST curves (x25519, ed25519, etc.). So, for the new policy we need to modify the expected fallback curve to NIST curve secp256r1.

I'll try to clarify this in the test code by adding a "NIST only fallback" array of policies and check for membership here.

Reply:
Also, I just noticed that [OQS just deprecated their fork of OpenSSL 1.1.1](https://github.com/open-quantum-safe/openssl/commit/98ce60974b53ac99b975cf2418aabab7519f1dff). I've created issue #4132 to track upgrading our test dependency to 3.0 and assigned it to myself.